### ISSUE TITLE
Hot Replacement Not Working

### ISSUE THREAD

**Issue by DmacMcgreg** [ducksoupdev/vue-webpack-typescript#11]
It seems that the hot replacement for modules is not working?
If I change the text in an html file, the whole application reloads, and I lose all the state. 
I'm using the default installation.

**Comment by ducksoupdev** [comment]
Thanks for raising the issue. Will look into it. Have you spotted an error in the setup/config for webpack?

**Comment by DmacMcgreg** [comment]
Hi @ducksoupdev ,
Thank you for responding and looking into it. Much appreciated. 
I'm definitely no expert in webpack, but I can continue to look into it.

**Comment by Stabzs** [comment]
@DmacMcgreg @ducksoupdev I'm not aware of html files having any sort of HMR for VueJS.  I believe HMR only applies to .vue components via vue-loader: https://vue-loader.vuejs.org/en/features/hot-reload.html

**Comment by cr0cK** [comment]
I confirm it doesn't work.

When I just update the text of a component:

```
[HMR] Cannot apply update. Need to do a full reload!
(anonymous) @ dev-server.js:33
dev-server.js:34 [HMR] Error: Aborted because 90 is not accepted
Update propagation: 90 -> 53 -> 54 -> 46 -> 131
    at hotApply (http://localhost:8080/main.js:432:30)
    at hotUpdateDownloaded (http://localhost:8080/main.js:285:13)
    at hotAddUpdateChunk (http://localhost:8080/main.js:265:13)
    at webpackHotUpdateCallback (http://localhost:8080/main.js:8:12)
    at http://localhost:8080/0.da78a1d428a5af062ca9.hot-update.js:1:1
```

**Comment by ducksoupdev** [comment]
The issue is as @cr0cK mentioned to do with the HMR and the hook into the change detection in order to reload the current components without browser refresh. The vue-loader module handles this using https://github.com/vuejs/vue-hot-reload-api. As the Typescript template is different, it requires a custom implementation to handle the reloading of components. This is a priority so will try and get a solution together as soon as possible.

**Comment by maccesch** [comment]
Any news on this?

**Comment by ducksoupdev** [comment]
@maccesch Apologies, things have been hectic and I haven't had time to work on it. It is a priority as I am using this approach at work and a fix is needed urgently :)

**Comment by maccesch** [comment]
@ducksoupdev I'd like to help you on this but I'm just a humble Webpack user without any knowledge of internals. Can you point me maybe in the right direction of what to do and what to read about?

**Comment by jordond** [comment]
Also hoping for an update.

**Comment by davidm-public** [comment]
doesn't work at all.
not sure what the point of another vue+typescript template is which:
a) can't do hot reloading, and
b) doesn't support .vue files
- why bother? seriously.

**Comment by ducksoupdev** [comment]
I'm sorry you feel that way @davidm-public. This style of using pure Typescript files rather than `.vue` files is an alternative way that plays nicely with editors and makes working with types easier.

**Comment by ducksoupdev** [comment]
I have been working on the hot reloading (apologies for the delay) and have a working app based on the current version of the template. I would welcome feedback on the following.

**`./hot-reload.ts`**

```ts
import Vue, { Component } from 'vue';
import * as api from 'vue-hot-reload-api';

const mode = process.env.ENV;

export const isHot = () => mode === 'development' && module.hot;

export function makeHot(id: string, component: Component, acceptFunc: void) {
  if (isHot) {
    api.install(Vue);
    if (!api.compatible) {
      throw new Error('vue-hot-reload-api is not compatible with the version of Vue you are using.');
    }

    api.createRecord(id, component);
    return true;
  }

  return false;
}

export const reload = api.reload;
```

**`./main.ts`**

```ts
...
import { isHot, makeHot, reload } from './hot-reload';
...

if (isHot) {
  makeHot('./components/navbar', NavbarComponent,
    module.hot.accept('./components/navbar', () => reload('./components/navbar', (<any>require('./components/navbar')).NavbarComponent)));
}

new Vue({
  el: '#app-main',
  router: createRouter(),
  components: {
    'navbar': NavbarComponent
  }
});
```

Getting it to play nicely with Typescript is tricky. The Webpack context is also hard to work with and cannot be passed into functions - it has to be used in the parent module context.

This is the minimal amount of code to enable hot-reloading.

I also have an updated `./router.ts` with routed components hot-reloading and can post that if required.

**Comment by ducksoupdev** [comment]
Example Vue app with hot reloading - https://github.com/ducksoupdev/vwt-hot-reload

**Comment by ducksoupdev** [comment]
Implemented in v1.1.0
